opds2_import.py

```diff
@@ -845,6 +876,28 @@ def extract_feed_data(self, feed, feed_url=None):
                 publication_metadata.primary_identifier.identifier
             ] = publication_metadata
 
+        node_finder = NodeFinder()
+
+        for error in parser_result.errors:
+            publication = node_finder.find_parent_or_self(
+                parser_result.root, error.node, opds2_ast.OPDS2Publication
+            )
+
+            if publication:
+                identifier = self._extract_identifier(publication)
+
+                if identifier:
```

Comment:
What happens to errors for which there is no publication or no identifier? It looks like it would be quietly ignored.

Reply:
What do you think would the right thing to do here? Should I log the error using the publication's title?

Reply:
A publication title would be helpful, if present; but it looks from the code like there might be cases where no pub is present.

I'm not sure what other information is available to the parser at this point. If you know line in file, entry count within file, or similar, those could be helpful (assuming the feed is relatively consistent).

Reply:
Maybe we just log an error with the publication? That might give enough info to whoever is digging through the logs to find the entry that is missing identifier information.

Reply:
@tdilauro, @jonathangreen 
The parser doesn't have information about line and character numbers, unfortunately. It traverses the object model resulting from parsing the JSON using `json.loads` and has access only to AST nodes and properties that contain those nodes:
```python
class BaseAnalyzerError(BaseError):
    """Exception raised in the case of any (syntax, semantic) errors thrown during parsing."""

    def __init__(self, node, node_property, message=None, inner_exception=None):
        """Initialize a new instance of BaseSemanticError class.

        :param node: AST node where the error was found
        :type node: webpub_manifest_parser.core.ast.Node

        :param node_property: AST node's property associated with the error
        :type node_property: Optional[webpub_manifest_parser.core.properties.Property]

        :param message: String containing description of the error occurred
        :type message: Optional[str]

        :param inner_exception: (Optional) Inner exception
        :type inner_exception: Optional[Exception]
        """
        super(BaseAnalyzerError, self).__init__(message, inner_exception)

        self._node = node
        self._node_property = node_property

    @property
    def node(self):
        """Return the AST node where the error was found.

        :return: Node where the error was found
        :rtype: webpub_manifest_parser.core.ast.Node
        """
        return self._node

    @property
    def node_property(self):
        """Return the AST node's property associated with the error.

        :return: AST node's property associated with the error
        :rtype: webpub_manifest_parser.core.properties.Property
        """
        return self._node_property
```

Reply:
However, most of the errors also have associated error messages:
```python
ODL_FEED_MISSING_PUBLICATIONS_SUBCOLLECTION_ERROR = partial(
    ManifestSemanticError,
    message="ODL feed '{0}' does not contain required 'publications' subcollection",
)

ODL_FEED_CONTAINS_REDUNDANT_GROUPS_SUBCOLLECTIONS_ERROR = partial(
    ManifestSemanticError,
    message="ODL feed '{0}' contains redundant 'groups' subcollections",
)

ODL_FEED_CONTAINS_REDUNDANT_FACETS_SUBCOLLECTIONS_ERROR = partial(
    ManifestSemanticError,
    message="ODL feed '{0}' contains redundant 'facets' subcollections",
)

ODL_FEED_CONTAINS_REDUNDANT_NAVIGATION_SUBCOLLECTION_ERROR = partial(
    ManifestSemanticError,
    message="ODL feed '{0}' contains redundant 'navigation' subcollection",
)

ODL_PUBLICATION_MUST_CONTAIN_EITHER_LICENSES_OR_OA_ACQUISITION_LINK_ERROR = partial(
    ODLPublicationSemanticError,
    message="ODL publication '{0}' contains neither 'licenses' subcollection nor "
    "an Open-Access Acquisition Link (http://opds-spec.org/acquisition/open-access)",
)

ODL_LICENSE_MUST_CONTAIN_SELF_LINK_TO_LICENSE_INFO_DOCUMENT_ERROR = partial(
    ODLLicenseSemanticError,
    message="ODL license '{0}' does not contain a 'self' link to the License Info Document",
)

ODL_LICENSE_MUST_CONTAIN_CHECKOUT_LINK_TO_LICENSE_STATUS_DOCUMENT_ERROR = partial(
    ODLLicenseSemanticError,
    message="ODL license '{0}' does not contain a 'checkout' link to the License Status Document",
)
```

We can log those error messages but they might not lead to a specific publication and it could be quite hard to find an error in a large feed.